### DISCUSSION TITLE
hi

### DISCUSSION THREAD

**Discussion by ekayembe** [mikro-orm/mikro-orm#2515] [Q&A]

**Comment by B4nan** [comment]
Yes it can, there are tests for this as well as some actual usage in example projects. Ideally provide full reproduction, often things can break if you have misconfigured base path or some other problem.

**Reply by B4nan** [reply]
Be sure to enable debug mode and try the `npx mikro-orm debug` command, it should give you some clues (note that CLI package needs to be installed locally to use it). 

One example of globs is in the realworld app (although that uses file globs, not dir globs, but the code that handles this is the same):

https://github.com/mikro-orm/nestjs-realworld-example-app/blob/master/src/mikro-orm.config.ts.example#L12

>  I am using a monorepo with lernajs and yarn workspaces---not sure if that has something to do with that.

That is quite possible, especially if the base path is configured as something dynamic.

Minimal repro would be very welcome, it can be easily a bug too. Or you could try to add some logging yourself, this is the place where it happens (globby is what supports globs, there is no explicit code for this in MikroORM):

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/metadata/MetadataDiscovery.ts#L115

**Reply by B4nan** [reply]
> And I've also been using the CLI and it doesn't expand the glob but keeps it in the url and says not found.

It should not expand, it should just check whether there are any matches for the path, and it does that and says there are no paths like that. Don't expect list of matches, that is not how it is supposed to work.

But in that screenshot it looks like you don't have `entitiesTs` set up? That is required for folder based discovery.

Btw this is what the realworld example gives for debug command:

```
➜  nestjs-realworld-example-app git:(v5) yarn mikro-orm debug
yarn run v1.22.15
$ /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/node_modules/.bin/mikro-orm debug
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 5.0.0-dev.495+5fe703238
   - node 16.10.0
   - knex 0.95.14
   - mysql2 not-found
   - typescript 4.5.2
 - package.json found
 - ts-node enabled
 - searched config paths:
   - /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/src/mikro-orm.config.ts (found)
   - /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/dist/mikro-orm.config.js (found)
   - /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/mikro-orm.config.ts (not found)
   - /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/mikro-orm.config.js (not found)
 - configuration found
 - will use `entities` array (contains 0 references and 1 paths)
   - /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/dist/**/*.entity.js (found)
 - could use `entitiesTs` array (contains 0 references and 1 paths)
   - /Users/adamek/htdocs/mikro-orm/nestjs-realworld-example-app/src/**/*.entity.ts (found)
✨  Done in 0.78s.
```
